File: CS/WindowsFormsApplication1/MyDockManager/MyWindowSkinPainter.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking.Paint;
using DevExpress.XtraBars.Styles;
using DevExpress.XtraBars.Docking;
using DevExpress.Utils;

namespace WindowsFormsApplication1
{
    public class MyWindowSkinPainter : WindowSkinPainter
    {
        public MyWindowSkinPainter(DockElementsSkinPainter painter) : base(painter) { }

        private static bool _DefaultDrawFirst = true;
        public static bool DefaultDrawFirst
        {
            get { return _DefaultDrawFirst; }
            set { _DefaultDrawFirst = value; }
        }

        private DockPanel _CurrentDockPanel;
        public DockPanel CurrentDockPanel
        {
            get { return _CurrentDockPanel; }
            set { _CurrentDockPanel = value; }
        }

        public bool RaiseCustomDraw(object args)
        {
            MyDockManager manager = CurrentDockPanel.DockManager as MyDockManager;
            MyCustomDrawArgs e = new MyCustomDrawArgs(args, CurrentDockPanel);
            if (manager != null)
            {
                manager.RaiseCustomDraw(e);
            }
            return e.Handled;
        }


        protected override void DrawWindowCaptionBackground(DrawWindowCaptionArgs args)
        {
            if (DefaultDrawFirst)
                base.DrawWindowCaptionBackground(args);
            if (!RaiseCustomDraw(args))
                base.DrawWindowCaptionBackground(args);
            
        }
        protected override void DrawApplicationCaptionBackgroud(DrawApplicationCaptionArgs args)
        {
            if (DefaultDrawFirst)
                base.DrawApplicationCaptionBackgroud(args);
            if (!RaiseCustomDraw(args)) 
                base.DrawApplicationCaptionBackgroud(args);
        }

        protected override void DrawCaptionText(DrawApplicationCaptionArgs e)
        {
            if (DefaultDrawFirst)
                base.DrawCaptionText(e);
            if (!RaiseCustomDraw(e))
                base.DrawCaptionText(e);
        }
    }
}

File: VB/WindowsFormsApplication1/MyDockManager/MyWindowSkinPainter.vb
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.XtraBars.Docking.Paint
Imports DevExpress.XtraBars.Docking

Namespace WindowsFormsApplication1

    Public Class MyWindowSkinPainter
        Inherits WindowSkinPainter

        Public Sub New(ByVal painter As DockElementsSkinPainter)
            MyBase.New(painter)
        End Sub

        Private Shared _DefaultDrawFirst As Boolean = True

        Public Shared Property DefaultDrawFirst As Boolean
            Get
                Return _DefaultDrawFirst
            End Get

            Set(ByVal value As Boolean)
                _DefaultDrawFirst = value
            End Set
        End Property

        Private _CurrentDockPanel As DockPanel

        Public Property CurrentDockPanel As DockPanel
            Get
                Return _CurrentDockPanel
            End Get

            Set(ByVal value As DockPanel)
                _CurrentDockPanel = value
            End Set
        End Property

        Public Function RaiseCustomDraw(ByVal args As Object) As Boolean
            Dim manager As MyDockManager = TryCast(CurrentDockPanel.DockManager, MyDockManager)
            Dim e As MyCustomDrawArgs = New MyCustomDrawArgs(args, CurrentDockPanel)
            If manager IsNot Nothing Then
                manager.RaiseCustomDraw(e)
            End If

            Return e.Handled
        End Function

        Protected Overrides Sub DrawWindowCaptionBackground(ByVal args As DrawWindowCaptionArgs)
            If DefaultDrawFirst Then MyBase.DrawWindowCaptionBackground(args)
            If Not RaiseCustomDraw(args) Then MyBase.DrawWindowCaptionBackground(args)
        End Sub

        Protected Overrides Sub DrawApplicationCaptionBackgroud(ByVal args As DrawApplicationCaptionArgs)
            If DefaultDrawFirst Then MyBase.DrawApplicationCaptionBackgroud(args)
            If Not RaiseCustomDraw(args) Then MyBase.DrawApplicationCaptionBackgroud(args)
        End Sub

        Protected Overrides Sub DrawCaptionText(ByVal e As DrawApplicationCaptionArgs)
            If DefaultDrawFirst Then MyBase.DrawCaptionText(e)
            If Not RaiseCustomDraw(e) Then MyBase.DrawCaptionText(e)
        End Sub
    End Class
End Namespace
